eos-tech-support/eos-stage-flatpak

```diff
@@ -68,5 +68,20 @@ for remote in `flatpak remote-list --system | awk {'print $1'}` ; do
         fi
         echo "Setting flatpak $remote URL to $new_url"
         flatpak remote-modify "$remote" --url="$new_url"
+
+        # Delete the appstream data and fetch it for the new remote;
+        # otherwise an update might be rejected as a downgrade
+        # https://github.com/flatpak/flatpak/issues/1615#issuecomment-385133763
+        rm -rf "$(dirname $flatpak_repo)/appstream/$remote"
+        for arch in `flatpak --supported-arches`; do
+            ostree --repo="$flatpak_repo" refs --delete "$remote:appstream/$arch"
+            # This might fail for the secondary architecture
+            flatpak update --appstream --arch="$arch" "$remote" >/dev/null 2>&1 || true
+        done
+
+        # Do the same with ostree-metadata
+        # https://github.com/flatpak/flatpak/issues/1477#issuecomment-372167938
+        ostree --repo="$flatpak_repo" refs --delete "$remote:ostree-metadata"
+        flatpak remote-ls "$remote" >/dev/null
```

Comment:
What is the purpose of calling `flatpak remote-ls` here?

Reply:
getting the new ostree-metadata. We could take it out and just let the next operation that needs it get it, and we could do the same with the appstream.

Reply:
I'm not sure how GNOME Software reacts to the appstream data changing while it's running, but better to have the right data than none I think.

Reply:
I'm not familiar with `flatpak remote-ls` updating the appstream data.  Don't you need `flatpak update --appstream` for that?

Reply:
Yes, that's why `flatpak update --appstream` is a few lines above that